addons/block_code/drag_manager/drag.gd

```diff
@@ -0,0 +1,197 @@
+@tool
+extends Control
+
+const Constants = preload("res://addons/block_code/ui/constants.gd")
+const DragManager = preload("res://addons/block_code/drag_manager/drag_manager.gd")
+const Types = preload("res://addons/block_code/types/types.gd")
+
+var _block: Block
+var _block_scope: String
+var _block_canvas: BlockCanvas
+var _preview_block: Control
+var _snap_points: Array[Node]
+var _delete_areas: Array[Rect2]
+var action: DragManager.DragAction:
```

Comment:
Since `DragAction` is only used in this class, you can move the `enum` here and avoid a circular dependency between `DragManager` and `Drag`.